lib/paperclip/color_extractor.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+require 'mime/types/columnar'
+
+module Paperclip
+  class ColorExtractor < Paperclip::Processor
+    MIN_CONTRAST        = 3.0
+    FREQUENCY_THRESHOLD = 0.01
+
+    def make
+      depth = 8
+
+      # Determine background palette by getting colors close to the image's edge only
+      background_palette = palette_from_histogram(convert(':source -alpha set -gravity Center -region \'75%x75%\' -alpha transparent +region -format %c -colors :quantity -depth :depth histogram:info:', source: File.expand_path(@file.path), quantity: 10, depth: depth), 10)
+      background_color   = background_palette.first
```

Comment:
In the edge case where all pixels in the image's edge are fully transparent, this would lead to `background_color` being `nil`, wouldn't it? The code should either return early, or use the `foreground_palette` in this case (provided it isn't empty itself).